src/main/java/seedu/duke/Duke.java

```diff
@@ -10,17 +11,24 @@
     private final Ui ui;
     private TaskList taskList;
     private LessonList lessonList;
+    private Parser parser;
 
     public Duke() {
         ui = new Ui();
         lessonList = new LessonList();
         taskList = new TaskList();
+        parser = new Parser();
     }
 
     public void startProgram() {
         boolean isExit = false;
         while (!isExit) {
             String userResponse = ui.readUserResponse();
+            try {
+                Command command = parser.parse(userResponse);
+            } catch (DukeException e) {
+                e.printStackTrace(); // error while parsing input
```

Comment:
Do note that `printStackTrace` is not exactly user-friendly. For development purpose it's ok, but we will have to remove it by the end our production.

Reply:
would it be better to use `getMessage()` instead?

Reply:
yep I'm leaving it as printStackTrace just until we have something better to print out the error

Reply:
Currently it's like [this](https://github.com/AY2122S1-CS2113T-W11-3/tp/blob/259469b72ea870b4de39afa44ec5fe2c810d6486/src/main/java/seedu/duke/Duke.java#L32). I think it's better if the printing is done inside the `Ui` class instead.